src/observers/link_prefetch_observer.js

```diff
@@ -78,12 +81,19 @@ export class LinkPrefetchObserver {
         )
 
         prefetchCache.setLater(location.toString(), fetchRequest, this.#cacheTtl)
-
-        link.addEventListener("mouseleave", () => prefetchCache.clear(), { once: true })
       }
     }
   }
 
+  #checkIfPrefetchValidAfterMouseLeave = (event) => {
+    if (event.target === this.#prefetchedLink) this.#cancelPrefetchRequest()
+  }
```

Comment:
I'd rename this to `#tryToCancelPrefetchRequest` and leave it in a single method:

```js
#tryToCancelPrefetchRequest = (event) => {
  if (event.target === this.#prefetchedLink) {
    prefetchCache.clear()
    this.#prefetchedLink = null
  }
}
```

Reason is that it's more evident that, on `mouseleave`, the callback is actually trying to cancel the prefetch request, rather than just "checking if it's valid".

Also, the reason I suggest adding "try", is because if the user doesn't leave the element within 100ms, we WILL make the prefetch request, and so this won't really cancel anything. So being explicit with "try" makes this more evident.

Reply:
`tryToCancelPrefetchRequest` also sound like you're going to try to cancel it, no matter what. But we only cancel if the user left the link. What about `cancelRequestIfObsolete`? I've implemented that in a9ec60d8e80462acb8862d8253492e2923bf6a7a.

Reply:
That sounds good, thank you!